modules/sievefilters/hm-sieve.php

```diff
@@ -2,10 +2,16 @@
 <?php
 
 class Hm_Sieve_Client_Factory {
-    public function init($user_config = null, $imap_account = null)
+    public function init($user_config = null, $imap_account = null, $is_nux_supported = false)
     {
         if ($imap_account && ! empty($imap_account['sieve_config_host'])) {
-            list($sieve_host, $sieve_port) = parse_sieve_config_host($imap_account['sieve_config_host']);
+            // Check if module nux is enabled and if it is, get the sieve host from the services
+            if($is_nux_supported && $sieve_config = get_sieve_host_from_services($imap_account['server'])) {
```

Comment:
Quite easier approach would be to store sieve config as part of imap server list and just use it here. Even if it is a quick service setup by nux module, make sure you always store the sieve host in the imap config. Then, you won't need to distinguish if it is a nux quick service one or normal imap.

Reply:
Well noted. However, in this context where @marclaporte  wanted that for instances with already configured servers, we handle the data structure change introduced in [this PR](https://github.com/cypht-org/cypht/pull/1199), it was necessary to find a way to connect Sieve using the structure according to the current version. For example, if in the current instance the format is `tls://`, then Sieve must automatically adapt to that.

I therefore relied on this logic to retrieve the data format from the NUX module, as it was also changed in [this PR](https://github.com/cypht-org/cypht/pull/1199).

The idea was that NUX would always provide the correct format regardless of the version. However, if I store these settings in an IMAP list, how can we avoid having to edit and reconfigure the servers on older instances? The goal is to format them automatically without user intervention.

Reply:
I believe we need a migration technique for this problem and other similar problems we are facing. Probably worth working on that in another PR as this one is urgent but the general idea is this:
- whenever a configuration format changes, we add a migration that will run on the old deployments
- we need to make sure this migration is run at each deployment - probably add it as part of the scripts/config_gen.php
- we can use configuration updates in a migration but also database schema updates (like the sqlite one)

Reply:
Copy that. I add a task for this